Update default behaviour of some agen-api methods. (#990)
livechat/livechat-public-docs: src/pages/messaging/agent-chat-api/changelog/index.mdx

  - [**threads_deleted**](/messaging/agent-chat-api/v3.4/rtm-reference/#threads_deleted)
- The **List Archives** ([Web](/messaging/agent-chat-api/v3.4/#list-archives) & [RTM](/messaging/agent-chat-api/v3.4/rtm-reference/#list-archives)) method:
  - has a new filter, `event_types`. It replaces the `events` filter.
  - has a new filter modifiers: `filters.agents.require_every_value`, `filters.tags.require_every_value`, `filters.sales.require_every_value`, `filters.goals.require_every_value`, and `filters.event_types.require_every_value`.
- Multiple methods no longer require the `chats--my:ro`, `chats--my:rw`, `chats.conversation--my:rw`, and `chats.conversation--my:rw` scopes.
- The **Start Chat** ([Web](/messaging/agent-chat-api/v3.4/#start-chat) & [RTM](/messaging/agent-chat-api/v3.4/rtm-reference/#start-chat)) and **Resume Chat** ([Web](/messaging/agent-chat-api/v3.4/#resume-chat) & [RTM](/messaging/agent-chat-api/v3.4/rtm-reference/#resume-chat)) methods now require that authors (except the requester) of all initial events are listed in the `users` field. Both the events and users have a fixed `visibility` set to `all`.

> - The **Deactivate Chat** ([Web](/messaging/agent-chat-api/v3.4/#deactivate-chat) & [RTM](/messaging/agent-chat-api/v3.4/rtm-reference/#deactivate-chat)) method now requires the user to be present in the chat's users list. This can be changed by setting a new param, `ignore_requester_presence`.

Reviewer: ```suggestion
- The **Deactivate Chat** ([Web](/messaging/agent-chat-api/v3.4/#deactivate-chat) & [RTM](/messaging/agent-chat-api/v3.4/rtm-reference/#deactivate-chat)) method now requires the requester to be present on the list of chat users. You can override it with a new param, `ignore_requester_presence`.
```

### Chats users

- Users now have a new field, `visibility`.
- The **Add User to Chat** ([Web](/messaging/agent-chat-api/v3.4/#add-user-to-chat) & [RTM](/messaging/agent-chat-api/v3.4/rtm-reference/#add-user-to-chat)) method:
  - now requires an active thread. The `require_active_thread` flag was removed.

>   - now requires the user to be present in the chat's users list. This can be changed by setting a new param, `ignore_requester_presence`.

Reviewer: same wording as for Deactve Chat

> - The **Remove User from Chat** ([Web](/messaging/agent-chat-api/v3.4/#remove-user-from-chat) & [RTM](/messaging/agent-chat-api/v3.4/rtm-reference/#remove-user-from-chat)) method now requires the user to be present in the chat's users list. This can be changed by setting a new param, `ignore_requester_presence`.

Reviewer: same wording as for Deactve Chat

### Chat access

- The **Transfer Chat** ([Web](/messaging/agent-chat-api/v3.4/#transfer-chat) & [RTM](/messaging/agent-chat-api/v3.4/rtm-reference/#transfer-chat)) method:
  - now requires an active thread.

>   - now requires the user to be present in the chat's users list. This can be changed by setting a new param, `ignore_requester_presence`.

Reviewer: same wording as for Deactve Chat

>   - the `force` parameter was renamed to `ignore_agents_availability`.

Reviewer: ```suggestion
  - had the `force` parameter renamed to `ignore_agents_availability`.
```

### Threads

- There are new properties: `previous_accessible_thread_id` and `next_accessible_thread_id`, which can be returned by the **List Archives** ([Web](/messaging/agent-chat-api/v3.4/#list-archives) & [RTM](/messaging/agent-chat-api/v3.4/rtm-reference/#list-archives)) method.

### Events

- The **Form** ([Web](/messaging/agent-chat-api/v3.4/#form) & [RTM](/messaging/agent-chat-api/v3.4/rtm-reference/#form)) and **Filled Form** ([Web](/messaging/agent-chat-api/v3.4/#filled-form) & [RTM](/messaging/agent-chat-api/v3.4/rtm-reference/#filled-form)) events have a new field, `form_type`.
- All **Events** ([Web](/messaging/agent-chat-api/v3.4/#events) & [RTM](/messaging/agent-chat-api/v3.4/rtm-reference/#events)) had the `recipients` field renamed to `visibility`.

> - The **Send Event** ([Web](/messaging/agent-chat-api/v3.4/#send-event) & [RTM](/messaging/agent-chat-api/v3.4/rtm-reference/#send-event)) method now requires the user to be present in the chat's users list.

Reviewer: same wording as for Deactve Chat
